### DISCUSSION TITLE
Seeking Assistance with SLF4J Logging Conflict in a Micronaut + Apache Beam Project

### DISCUSSION THREAD

**Discussion by glats** [micronaut-projects/micronaut-core#10430] [Q&A]
Hello Micronaut Community,

I'm reaching out for assistance with an issue in my Maven project that uses Micronaut 4 and Apache Beam. While this may not be a direct issue with Micronaut, I believe the expertise in this community could provide valuable insights.

I'm facing a logging issue where, despite including `logback-classic` in my dependencies, I encounter the standard SLF4J error about missing `StaticLoggerBinder`. The problem persists even after excluding `org.slf4j` from various dependencies. Interestingly, removing `beam-sdks-java-io-google-cloud-platform` resolves the logging issue, indicating a potential conflict. slf4j-simple also appears in the classpath, which might be part of the problem.

The project is accessible at https://github.com/j1cs/logback-error, and it can be run with `./mvnw mn:run -DskipTests=true`. The expected log output should be:

```
00:14:57.306 [main] INFO  i.m.c.DefaultApplicationContext$RuntimeConfiguredEnvironment - Established active environments: [cli]
00:14:57.308 [main] INFO  i.m.c.DefaultApplicationContext$BootstrapEnvironment - Established active environments: [cli]
00:14:57.328 [main] INFO  i.m.context.DefaultBeanContext - Reading bootstrap environment configuration
```
Any guidance on resolving this SLF4J implementation conflict or suggestions to debug this further would be greatly appreciated.

More info: https://stackoverflow.com/questions/77903737/conflict-with-slf4j-implementations-in-maven-project-with-micronaut-and-apache-b

Thank you!

**Comment by nedelva** [comment]
It looks to me that _micronaut-maven-plugin_ has a transitive dependency on _slf4j-simple_, version **1.7.36**. 
Same `mn:run` command but with Maven's output level set to DEBUG will show that:

> [DEBUG] Importing foreign packages into class realm plugin>io.micronaut.maven:micronaut-maven-plugin:4.0.8
> [DEBUG]   Imported:  < project>me.jics:app:0.1
> [DEBUG] Populating class realm plugin>io.micronaut.maven:micronaut-maven-plugin:4.0.8
> [DEBUG]   Included: io.micronaut.maven:micronaut-maven-plugin:jar:4.0.8
> [DEBUG]   Included: io.micronaut.maven:micronaut-maven-core:jar:4.0.8
> [DEBUG]   Included: io.micronaut:micronaut-inject:jar:4.0.3
> [DEBUG]   Included: jakarta.inject:jakarta.inject-api:jar:2.0.1
> [DEBUG]   Included: jakarta.annotation:jakarta.annotation-api:jar:2.1.1
> [DEBUG]   Included: io.micronaut:micronaut-core:jar:4.0.3
> .... (many lines removed)
> [DEBUG]   Included: com.github.jknack:handlebars:jar:4.2.1
> [DEBUG]   Included: com.github.jknack:handlebars-jackson2:jar:4.2.1
> [DEBUG]   Included: org.slf4j:slf4j-ext:jar:1.7.36
> [DEBUG]   Included: org.slf4j:slf4j-simple:jar:1.7.36
> ..... (etc etc)
> [DEBUG]   Included: com.google.errorprone:error_prone_annotations:jar:2.10.0

If you launch instead the _AppCommand_ via the java command, the output is as you expected it to be:
`java.exe -Dfile.encoding=UTF-8 -classpath %MODULE_CLASSPATH% me.jics.AppCommand`
will produce 

> 11:33:14.828 [main] INFO  i.m.c.DefaultApplicationContext$RuntimeConfiguredEnvironment - Established active environments: [cli]
> 11:33:14.833 [main] INFO  i.m.c.DefaultApplicationContext$BootstrapEnvironment - Established active environments: [cli]
> 11:33:14.867 [main] INFO  i.m.context.DefaultBeanContext - Reading bootstrap environment configuration

Hope this helps.

**Reply by glats** [reply]
so the problem is with the micronaut-maven-plugin. i wonder if i should report this :thinking:

**Reply by graemerocher** [reply]
if you provide a reproducer in an issue we can take a look, I doubt it is the maven plugin though as that would indicate a much bigger general problem

**Reply by glats** [reply]
@graemerocher as you can see in the original question i provided a sample project with the error. now with the evidence provided by @nedelva maybe i can open an issue with this problem. what do you think?
